app/src/main/java/com/door43/translationstudio/tasks/io/Request.java

```diff
@@ -0,0 +1,131 @@
+package com.door43.translationstudio.tasks.io;
+
+import android.util.Base64;
+
+import org.unfoldingword.gogsclient.Response;
+import org.unfoldingword.gogsclient.User;
+import org.unfoldingword.tools.logger.Logger;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import javax.net.ssl.HttpsURLConnection;
+
+public class Request {
```

Comment:
This class is extracted from the Gogs API source code

Reply:
Is this code from an existing library?  If so, we need to make sure that it's OSS and that we point to the source.

Reply:
I see the open source license from the source repo from which I extracted the request method. 
Also, I could use OkHttp library for making request if you prefer that.

https://github.com/unfoldingWord-dev/android-gogs-client/blob/master/gogs-client/src/main/java/org/unfoldingword/gogsclient/GogsAPI.java#L79

Reply:
Of course I'm not a lawyer, but the wording on their license makes it seem as though it can be used for whatever purposes. It's available here: https://github.com/unfoldingWord-dev/android-gogs-client/blob/master/LICENSE

It seems that, due to this line in the license - "
The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software." - we can use it so long as we add the license to the file that is using the software.

Though, again, I'm not sure that a developer is qualified to make that decision. Rewriting things with OkHttp may be simpler both from a development and a licensing point of view.

Reply:
@PurpleGuitar here's the OkHttp implementation that I mentioned earlier.